lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -152,10 +153,13 @@ export class SilentFlowClient extends BaseClient {
      */
     private isTokenExpired(expiresOn: string): boolean {
         // check for access token expiry
-        const expirationSec = Number(expiresOn);
+        let expirationSec = Number(expiresOn);
         const offsetCurrentTimeSec = TimeUtils.nowSeconds() + this.config.systemOptions.tokenRenewalOffsetSeconds;
 
         // Check if refresh is forced, or if tokens are expired. If neither are true, return a token response with the found token entry.
-        return (expirationSec && expirationSec > offsetCurrentTimeSec);
+        if (!expirationSec) {
+            expirationSec = 0;
```

Comment:
This could be done when you create `expirationSec`: `const expirationSec = Number(expiresOn) || 0;`

Reply:
done.